codegen/src/main/resources/software/amazon/awssdk/codegen/rules/AuthSchemeUtils.java.resource

```diff
@@ -50,56 +55,71 @@ public final class AuthSchemeUtils {
 
-                    Value signingName = scheme.get(Identifier.of("signingName"));
-                    if (signingName != null) {
-                        schemeBuilder.signingName(signingName.expectString());
-                    }
+                authSchemes.add(schemeBuilder.build());
+            }
+                break;
+            case SIGV4_NAME: {
+                SigV4AuthScheme.Builder schemeBuilder = SigV4AuthScheme.builder();
 
-                    Value signingRegion = scheme.get(Identifier.of("signingRegion"));
-                    if (signingRegion != null) {
-                        schemeBuilder.signingRegion(signingRegion.expectString());
-                    }
+                Value signingName = scheme.get(Identifier.of("signingName"));
+                if (signingName != null) {
+                    schemeBuilder.signingName(signingName.expectString());
+                }
 
-                    Value disableDoubleEncoding = scheme.get(Identifier.of("disableDoubleEncoding"));
-                    if (disableDoubleEncoding != null) {
-                        schemeBuilder.disableDoubleEncoding(disableDoubleEncoding.expectBool());
-                    }
+                Value signingRegion = scheme.get(Identifier.of("signingRegion"));
+                if (signingRegion != null) {
+                    schemeBuilder.signingRegion(signingRegion.expectString());
+                }
 
-                    authSchemes.add(schemeBuilder.build());
+                Value disableDoubleEncoding = scheme.get(Identifier.of("disableDoubleEncoding"));
+                if (disableDoubleEncoding != null) {
+                    schemeBuilder.disableDoubleEncoding(disableDoubleEncoding.expectBool());
                 }
+
+                authSchemes.add(schemeBuilder.build());
+            }
+                break;
+            default:
+                LOG.debug(() -> "Ignoring unknown auth scheme: " + authSchemeName);
                 break;
-                default:
-                    LOG.debug(() -> "Ignoring unknown auth scheme: " + authSchemeName);
-                    break;
             }
         }
         return authSchemes;
     }
+
+    public static <T extends Identity> void putSelectedAuthScheme(ExecutionAttributes attributes,
+                                                                  SelectedAuthScheme<T> selectedAuthScheme) {
+        SelectedAuthScheme<?> existingAuthScheme = attributes.getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
+        if (existingAuthScheme != null) {
+            AuthSchemeOption.Builder selectedOption = selectedAuthScheme.authSchemeOption().toBuilder();
+            existingAuthScheme.authSchemeOption().forEachIdentityProperty(selectedOption::putIdentityPropertyIfAbsent);
+            existingAuthScheme.authSchemeOption().forEachSignerProperty(selectedOption::putSignerPropertyIfAbsent);
```

Comment:
the putIfAbsent here means if the selectedAuthScheme overwrites the existingAuthScheme's values - this seems fine. IIUC, the only case where existingAuthScheme would be non-null when it reaches this code path would be for attributes set [here](https://github.com/aws/aws-sdk-java-v2/blob/12e3994c75b2858cbd42ad0a2b35d7384e3b1f3e/core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/AwsExecutionContextBuilder.java#L72-L116) before calling the AuthSchemeInterceptor (which calls this AuthSchemeUtils) [few lines later here](https://github.com/aws/aws-sdk-java-v2/blob/12e3994c75b2858cbd42ad0a2b35d7384e3b1f3e/core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/AwsExecutionContextBuilder.java#L126).

Though makes me wonder if this method should have comments on appropriate use and prevent it getting used in some other context. Not a big deal I think, though probably another reason to move this method inside AuthSchemeInterceptor.